packages/battery/battery_linux/lib/battery_linux.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+import 'dart:async';
+
+import 'package:battery_platform_interface/battery_platform_interface.dart';
+
+/// The linux implementation of [BatteryPlatform]
+///
+/// This class implements the `package:battery` functionality for linux
+class BatteryLinux extends BatteryPlatform {
+  /// Registers this class as the default instance of [BatteryPlatform]
+  static void register() {
+    BatteryPlatform.instance = BatteryLinux();
+  }
+
+  /// Stores the path to battery info directory.
+  /// This can be BAT0 or BAT1.
+  String _path;
+
+  /// Gets the path of battery info directory.
+  Future<void> get _getBatteryPath async {
+    _path = "/sys/class/power_supply/";
+    Directory _directory = Directory("/sys/class/power_supply/BAT0");
```

Comment:
Would it be safer to enumerate the subdirectories of `/sys/class/power_supply` and grab one from there (or maybe control multiple BATx if more than one exists?)

Reply:
It means we have to return a List of `batteryData` to display info of all the batteries.

Reply:
I think a better information is stated [here](https://askubuntu.com/a/309146/1050100). What's your call on this @stuartmorgan .

Reply:
I'm not sure what your question is exactly, but I would not expect an answer to the question "How do I check the battery's status **via the terminal**?" to be a good source of information about how the best way to programmatically check, and monitor, battery information.